File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/item/ItemTagDao.java
package com.legue.axel.lolsummonertool.database.dao.item;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.legue.axel.lolsummonertool.database.model.item.ItemTag;

import java.util.List;

@Dao
public interface ItemTagDao {

    @Query("SELECT * FROM item_tags ORDER BY id")
    LiveData<List<ItemTag>> getItemTags();

    @Query("SELECT * FROM item_tags WHERE id = :itemTagId")
    LiveData<ItemTag> getItemTagById(int itemTagId);

    @Insert
    void insertItemTag(ItemTag itemTag);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllItemTag(List<ItemTag> itemTags);

    @Delete
    void deleteItemTag(ItemTag itemTag);

    @Query("DELETE FROM item_tags")
    void deleteAll();

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateItemTag(ItemTag itemTag);
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/item/ItemTagDao.kt
package com.legue.axel.lolsummonertool.database.dao.item

import androidx.lifecycle.LiveData
import androidx.room.*
import com.legue.axel.lolsummonertool.database.model.item.ItemTag

@Dao
interface ItemTagDao {

    @Query("SELECT * FROM item_tags ORDER BY id")
    fun getItemTags(): LiveData<List<ItemTag>>

    @Query("SELECT * FROM item_tags WHERE id = :itemTagId")
    fun getItemTagById(itemTagId: Int): LiveData<ItemTag>

    @Insert
    fun insertItemTag(itemTag: ItemTag)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllItemTag(itemTags: List<ItemTag>)

    @Delete
    fun deleteItemTag(itemTag: ItemTag)

    @Query("DELETE FROM item_tags")
    fun deleteAll()

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateItemTag(itemTag: ItemTag)
}
